### PR TITLE
Add support for custom per-volume uniforms

### PR DESCRIPTION
**Pull request by skalarproduktraum**
This PR adds:
* support for adding custom per-volume uniforms
* support for integer array uniform types (uniform1i, 2i,...)
* support for Vector4f array type

### CODE REVIEW

File: src/main/java/tpietzsch/example2/MultiVolumeShaderMip.java [hunk 1]
```diff
@@ -29,16 +29,15 @@
...
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.function.BiConsumer;
 import net.imglib2.type.numeric.ARGBType;
-import org.joml.Matrix4f;
-import org.joml.Matrix4fc;
-import org.joml.Vector2f;
-import org.joml.Vector4f;
+import org.joml.*;
```
Comment (imagejan) [inline src/main/java/tpietzsch/example2/MultiVolumeShaderMip.java]: http://www.javadude.com/posts/20040522-import-on-demand-is-evil/

Comment (skalarproduktraum) [reply src/main/java/tpietzsch/example2/MultiVolumeShaderMip.java]: Whoops, I had totally missed that. Thanks @imagejan, will fix 👍

Comment (moreApi) [reply src/main/java/tpietzsch/example2/MultiVolumeShaderMip.java]: fixed

**Comment by skalarproduktraum** [comment]
Poke @tpietzsch :-)

**Comment by skalarproduktraum** [comment]
Poke again @tpietzsch :-)

**Comment by skalarproduktraum** [comment]
And one more poke for @tpietzsch :-) We'd really appreciate if this could be merged soonish 👍

**Comment by skalarproduktraum** [comment]
Apologies I missed the July poke, please regard this as a double-poke then 🤣

**Comment by ctrueden** [comment]
@tpietzsch scenery needs this in order to stop forking. Let's work on it during this week's hackathon! :+1:

**Comment by tpietzsch** [comment]
Yes 👍 
This is one of the last things to do before I can release...

**Comment by tpietzsch** [comment]
I kind of merged this via https://github.com/bigdataviewer/bigvolumeviewer-core/commit/9c6694235b6abbbd07745b53e53b6936a3437220

I reworked it heavily in https://github.com/bigdataviewer/bigvolumeviewer-core/commit/7434f7670be0572c40090aa7fcc0dcce564c3bbb though.

I removed the `additionalSamplers` and `additionalUniforms` HashMaps again, because the `Uniform*` classes already represent a layer of caching that defers the "actual setting" via the GpuContext to  later.
The logic to decide by type of the `Object value` argument what uniform type to set moved to `SegmentedShader`.

Everything should now done by just `MultiVolumeShaderMip.setUniform(...)`.
The methods
```
	public void registerCustomSampler( int index, String name, Texture texture )
	public void setCustomUniformForVolume( int index, String name, Object value )
	public void removeCustomUniformForVolume( int index, String name )
	public void setCustomIntArrayUniformForVolume( int index, String name, final int elementSize, final int[] value )
	public void setCustomFloatArrayUniformForVolume( int index, String name, final int elementSize, final float[] value )
```
I left in as `@Deprecated`, but will remove in the next release. I hope that makes it easier to first migrate scenery to the changed package names, and then -- if everything works -- replace the deprecated methods.

***The only thing that is not covered is `removeCustomUniformForVolume`***
I could add that if it is still needed. But I don't really understand why it would be useful. Either the uniform exists in the program or not, so if it is removed (instead of just overridden), wouldn't that mean that the program doesn't work anymore???

**Pull request closed without merging**